modules/gapi/src/executor/gstreamingexecutor.cpp

```diff
@@ -497,6 +525,7 @@ void emitterActorThread(std::shared_ptr<cv::gimpl::GIslandEmitter> emitter,
                 || cv::util::holds_alternative<Stop>(cmd));
     if (cv::util::holds_alternative<Stop>(cmd))
     {
+        GAPI_LOG_INFO(nullptr, "Early stop for: " << emitter->name() << " detected");
```

Comment:
Troubleshooting  & Logging BKMs:

Log incoming event up right before process it (possible stuck on push())